tests/Collection/CollectionFunctionalTest.php

```diff
@@ -311,6 +311,71 @@ public function testMapReduce()
         $this->assertNotEmpty($result->getCounts());
     }
 
+    public function testDontInheritWriteConcernInTransaction()
+    {
+        $this->skipIfTransactionsAreNotSupported();
+
+        $this->createCollection();
+
+        $session = $this->manager->startSession();
+        $session->startTransaction([
+            'readConcern' => new \MongoDB\Driver\ReadConcern("snapshot"),
+            'writeConcern' => new \MongoDB\Driver\WriteConcern(\MongoDB\Driver\WriteConcern::MAJORITY)
+        ]);
+
+        $collection = new Collection(
+            $this->manager,
+            $this->getDatabaseName(),
+            $this->getCollectionName(),
+            [ 'writeConcern' => new \MongoDB\Driver\WriteConcern(1) ]
+        );
+
+        (new CommandObserver)->observe(
+            function() use ($collection, $session) {
+                $collection->insertOne(
+                    ['test' => 'foo'],
+                    [
+                        'session' => $session,
+                    ]
+                );
+            },
+            function(array $event) {
+                $this->assertObjectNotHasAttribute('writeConcern', $event['started']->getCommand());
+            }
+        );
+
+        $this->dropCollection();
+    }
```

Comment:
Would it be prudent to call `$session->endSession();` at the end of these functions? I recall that coming up when I first added the aggregate/find transaction tests. See https://github.com/mongodb/mongo-php-library/pull/564#discussion_r202395078 for more context.

This is particularly necessary for tests that throw, since the exception's stack trace ends up holding a reference to the session variable, which prevents it from being closed and terminating the transaction. If we don't clean up the session, I believe future test methods could block unnecessarily.

A `try { ... } finally { $session->endSession(); }` should suffice for now. Long term, I think we can come up with a better solution for this. I'm experimenting with something in my work for SPEC-1133 to track _all_ collections used during a test (e.g. output collections) for cleanup during `tearDown()`, instead of only having `tearDown()` drop the single collection under test. If that works out, I can see us doing something similar to track sessions used during a test to ensure that `tearDown()` calls `endSession()` for us.

Reply:
Created [PHPLIB-399](https://jira.mongodb.org/browse/PHPLIB-399) to record the idea about tracking sessions.